src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -2593,35 +2752,64 @@ public class Derived : Mid
+        string x2 = m.M();
+        string x3 = d.M();
+    }
+}
+";
 
             var references = new[] { ref0, ref1a };
-            var comp = CreateCompilationWithoutCovariantReturns(s2, references).VerifyDiagnostics(
+            var comp = CreateCompilationWithCovariantReturns(s2, references, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
                 // (4,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
                 //     public override string M() => null;
                 Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(4, 28)
                 );
-            verify1(comp);
+            verify1(SourceView(comp, assignments1));
 
             comp = CreateCompilationWithCovariantReturns(s2, references).VerifyDiagnostics(
                 );
-            verify1(comp);
-            verify1(CompilationReferenceView(comp, assignments, references));
-            verify1(MetadataView(comp, assignments, references));
-            verify1(RetargetingView(comp, assignments, references));
+            verify1(SourceView(comp, assignments1));
+            verify1(CompilationReferenceView(comp, assignments1, references));
+            verify1(MetadataView(comp, assignments1, references));
+            verify1(RetargetingView(comp, assignments1, references));
 
             references = new[] { ref0, ref1b };
             // we do not test CompilationReferenceView because the changed reference would cause us to retarget
-            verify2(MetadataView(comp, assignments, references));
-            verify2(RetargetingView(comp, assignments, references));
+            verify2(MetadataView(comp, assignments2, references));
 
-            static void verify1(CSharpCompilation comp)
+            // The following error probably arises from the fact that Mid.M and Derived.M are unrelated (they both override Base.M directly).
+            // See https://github.com/dotnet/roslyn/issues/45798
+            verify2(RetargetingView(comp, assignments2, references, expectedDiagnostics:
+                // (8,23): error CS0121: The call is ambiguous between the following methods or properties: 'Base.M()' and 'Base.M()'
+                //         string x3 = d.M();
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("Base.M()", "Base.M()").WithLocation(8, 23)
+                ));
+
+            static void verify1(CSharpCompilation comp, bool verifyAssignments = true)
             {
+                if (verifyAssignments)
+                    VerifyAssignments(comp, 3);
                 VerifyOverride(comp, "Derived.M", "System.String Derived.M()", "System.Object Base.M()");
             }
 
-            static void verify2(CSharpCompilation comp)
+            static void verify2(CSharpCompilation comp, params DiagnosticDescription[] expectedDiagnostics)
```

Comment:
>expectedDiagnostics [](start = 87, length = 19)

It looks like parameter is not used

Reply:
See the end of line 2799.

---
In reply to: [452423561](https://github.com/dotnet/roslyn/pull/45824#discussion_r452423561) [](ancestors = 452423561)